perl.c

```diff
@@ -1350,8 +1350,11 @@ perl_destruct(pTHXx)
 
     /* constant strings */
     for (i = 0; i < SV_CONSTS_COUNT; i++) {
-        SvREFCNT_dec(PL_sv_consts[i]);
-        PL_sv_consts[i] = NULL;
+        SV * sv = PL_sv_consts[i];
+        if(sv) {
+            PL_sv_consts[i] = NULL;
+            SvREFCNT_dec_NN(sv);
```

Comment:
With gcc-12.2.0. the ordering of these has an impact. This way around, `perl_destruct` is 16 instructions larger, where as with:
```
        if(sv) {
            SvREFCNT_dec_NN(sv);
            PL_sv_consts[i] = NULL;
        }
```
the function size is unchanged. (I haven't checked to see if the entire function is identical.)

The ordering makes no difference on clang-11. Both are 16 instructions smaller than the current code.

I dunno what you see with MSVC?

Reply:
Can you copy paste the GCC cmd line flags, especially your version of GCC, cpu arch, and -O flag? I'd like to try to reproduce what your seeing, and think over a couple things in my head or possible suspiciousions of what could happen and what does happen.